ooniprobe/Utility/OONIApi.m

```diff
@@ -3,16 +3,19 @@
 #import "SettingsUtility.h"
 #import "NetworkSession.h"
 #import "Url.h"
+#import "VersionUtility.h"
 
 @implementation OONIApi
 
 + (void)downloadUrls:(void (^)(NSArray*))successcb onError:(void (^)(NSError*))errorcb {
-    id<GeoIPLookupTask> task = [Engine geoIPLookupTask];
-    [task setTimeout:DEFAULT_TIMEOUT];
-    id<GeoIPLookupResults> results = [task perform];
-    NSString *cc = @"XX";
-    if ([results isGood])
-        cc = [results probeCC];
+    NSError *error;
+    NSString *cc = [Engine resolveProbeCCWithSoftwareName:SOFTWARE_NAME
+                                          softwareVersion:[VersionUtility get_software_version]
+                                                  timeout:DEFAULT_TIMEOUT
+                                                    error:&error];
+    if (error != nil || cc == nil) {
```

Comment:
In this context it's sufficient to check for one of them, not both. I'd vote for error:

```suggestion
    if (error != nil) {
```

Reply:
are we sure about it?
if cc is null it would crash big times the app, I prefer the double check as cc doesn't have a default value

Reply:
So, I'd rather check for just `cc` then! (All of this is complex just because we don't have union types 😢)